firebase-firestore/src/testUtil/java/com/google/firebase/firestore/TestAccessHelper.java

```diff
@@ -28,4 +29,14 @@ public static DocumentReference createDocumentReference(DocumentKey documentKey)
   public static DocumentKey referenceKey(DocumentReference documentReference) {
     return documentReference.getKey();
   }
+
+  /**
+   * Install mocks for `FirebaseFirestore.getFirestoreSettings()` and
+   * `FirebaseFirestore.getUserDataWriter()`, which are used in DocumentSnapshot tests.
+   */
+  public static void installDocumentSnapshotMocks(FirebaseFirestore firestore) {
```

Comment:
This kind of centralization of mocking creation makes me uneasy. It seems like we're using the wrong tool for the job. If `TestUtil.firestore()` returned a real instance of Firestore these tests wouldn't have to mock anything to get what's otherwise default behavior of the instance.

Also, the function of this particular class is to expose as public methods things that are otherwise access protected in the public API. A way to allow mocking of package private methods would be just to contralize the `Mockito.when` call like this:


```java
  public static OngoingStubbing<UserDataReader> getUserDataWriter(FirebaseFirestore firestore) {
    return Mockito.when(firestore.getUserDataWriter());
  }
```

Reply:
`installDocumentSnapshotMocks()` is not needed in our codebase to bypass access restrictions, but it was in the Kotlin tests since they live in a subpackage.

Either way, I reverted this change as I removed the getUserDataWriter() instance method on FirebaseFirestore.